packages/grid/x-data-grid-pro/src/DataGridProVirtualScroller.tsx

```diff
@@ -128,6 +159,14 @@ const DataGridProVirtualScroller = React.forwardRef<
     }
   };
 
+  const getRowProps = (id: GridRowId) => {
+    if (!expandedRowIds.includes(id)) {
```

Comment:
Maybe `expandedRowIds` should be an object like we discussed for the visible columns.
To avoid this linear complexity `includes` which can cause problem for very large datasets.
What would be the con of a `Record<GridRowId, boolean>` for the model ?

Reply:
I don't have the bandwidth to update everything now, maybe in v6. I think `expandedRowIds.includes` is fine because not many rows will be in an open state at the same time.

Reply:
If you think the use case where the user expand all detail panels of all pages is not something realistic, then I agree that it is negligeable.

We can wait for v6 and user feedbakcs, I'm closing this one :+1: